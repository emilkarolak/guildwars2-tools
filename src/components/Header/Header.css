nav {
  font-size: 1.2rem;
  height: 60px;
}

nav ul {
  font-weight: bold;
}