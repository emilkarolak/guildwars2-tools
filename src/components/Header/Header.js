import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

export default () => (
  <nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
    <a className="navbar-brand" href="/">Guild Wars 2 - Tools</a>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation"><span className="navbar-toggler-icon"></span></button>
    <div className="collapse navbar-collapse" id="navbarCollapse">
      <ul className="navbar-nav mr-auto">
        <li className="nav-item"><NavLink to="/" exact className="nav-link" activeClassName="active">Map</NavLink></li>
        <li className="nav-item"><NavLink to="/account" className="nav-link" activeClassName="active">Account</NavLink></li>
      </ul>
    </div>
  </nav>
)