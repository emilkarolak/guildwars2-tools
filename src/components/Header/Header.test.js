import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

it('renders without crashing', () => {
  shallow(<BrowserRouter><Header /></BrowserRouter>);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Header /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders navigation menu', () => {
  const wrapper = shallow(<Header />);
  const element = <ul className="navbar-nav mr-auto">
    <li className="nav-item"><NavLink to="/" exact className="nav-link" activeClassName="active">Map</NavLink></li>
    <li className="nav-item"><NavLink to="/account" className="nav-link" activeClassName="active">Account</NavLink></li>
  </ul>;
  expect(wrapper).toContainReact(element);
});