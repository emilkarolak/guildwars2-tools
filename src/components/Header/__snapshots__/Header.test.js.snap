// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Guild Wars 2 - Tools
  </a>
  <button
    aria-controls="navbarCollapse"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarCollapse"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarCollapse"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Map
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/account"
          onClick={[Function]}
        >
          Account
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
